core/src/main/java/org/moditect/jfrunit/JfrEventType.java

```diff
@@ -0,0 +1,178 @@
+        return Objects.equals(recordedEvent.getInstant(name), expected);
+    }
+
+    protected static boolean matches(Duration expected, String name, RecordedEvent recordedEvent) {
+        return Objects.equals(recordedEvent.getDuration(name), expected);
+    }
+
+    protected static boolean matches(Long expected, String name, RecordedEvent recordedEvent) {
+        return Objects.equals(recordedEvent.getLong(name), expected);
+    }
+
+    protected static boolean matches(Integer expected, String name, RecordedEvent recordedEvent) {
+        return Objects.equals(recordedEvent.getInt(name), expected);
+    }
+
+    protected static boolean matches(Boolean expected, String name, RecordedEvent recordedEvent) {
+        return Objects.equals(recordedEvent.getBoolean(name), expected);
+    }
+
+    protected static boolean matches(Byte expected, String name, RecordedEvent recordedEvent) {
+        return Objects.equals(recordedEvent.getByte(name), expected);
+    }
+
+    protected static boolean matches(Character expected, String name, RecordedEvent recordedEvent) {
+        return Objects.equals(recordedEvent.getChar(name), expected);
+    }
+
+    protected static boolean matches(Double expected, String name, RecordedEvent recordedEvent) {
+        return Objects.equals(recordedEvent.getDouble(name), expected);
+    }
+
+    protected static boolean matches(Float expected, String name, RecordedEvent recordedEvent) {
+        return Objects.equals(recordedEvent.getFloat(name), expected);
+    }
+
+    protected static boolean matches(Short expected, String name, RecordedEvent recordedEvent) {
+        return Objects.equals(recordedEvent.getShort(name), expected);
+    }
+
+    protected static boolean matches(String expected, String name, RecordedEvent recordedEvent) {
+        return Objects.equals(recordedEvent.getString(name), expected);
+    }
+
+    protected static boolean matches(ExpectedThread expected, String name, RecordedEvent recordedEvent) {
+        return expected.test(recordedEvent.getThread(name));
+    }
+
+    protected static boolean matches(ExpectedClass expected, String name, RecordedEvent recordedEvent) {
+        return expected.test(recordedEvent.getClass(name));
+    }
+
+    protected static boolean matches(ExpectedStackFrame expected, String name, RecordedEvent recordedEvent) {
+        RecordedStackTrace stackTrace = recordedEvent.getStackTrace();
+        if (stackTrace == null) {
+            return false;
+        }
+        List<RecordedFrame> recordedFrames = stackTrace.getFrames();
+        if (recordedFrames.stream().noneMatch(expected::test)) {
+            return false;
+        }
+
+        return true;
+    }
+
+    protected static boolean matches(ExpectedClassLoader expected, String name, RecordedEvent recordedEvent) {
+        throw new UnsupportedOperationException("matches ExpectedClassLoader not implemented");
```

Comment:
Are you still planning to do this as part of this PR? Or as a follow-up? Let's log a separate issue in this case.